server/services/transaction.service.js

```diff
@@ -67,6 +69,14 @@ export const addTransaction = async (transaction) => {
       throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
     });
 
+    const referral = await activatePendingUserReferral(offer);
```

Comment:
Hmmm! I think this is also wrong. We only need to `activatePendingUserReferral(offer)` when a user accepts an offer (you would need this for your test or simply add the offerId).

The right thing to do here is to get Referral by OfferId

```suggestion
    const referral = await getReferralByOfferId(offer);
```